### DISCUSSION TITLE
Specify custom discount rates per hour

### DISCUSSION THREAD

**Discussion by GuyKh** [GuyKh/iec-custom-component#121] [Ideas]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/ludeeus/iec/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

No.

### Describe the solution you'd like

With private electric providers, many offer a discount per hours - e.g. _15% off on weekdays between 08:00-17:00_.
Need to think about how to model, and use it within the component to get a more clear estimation of the cost.

I've thought about a json with hour ranges describing the discount from the full tariff, but that's a longshot.

e.g.
```
{
    // Sunday
    "0": { 
        "from": "08:00",
        "to": "17:00",
        "price": 0.50
    },
    "1": { 
        "from": "20:00",
        "to": "23:59",
        "discountPercent": 15
    },
    "2": { 
        "from": "00:00",
        "to": "08:08",
        "discountPercent": 15
    }
    // ...
}
```

Downside is that it's not very intuitive, and hard to maintain.

**Comment by OnFreund** [comment]
HA has [existing mechanisms](https://www.home-assistant.io/integrations/utility_meter/#advanced-configuration) for this.

**Comment by GuyKh** [comment]
@OnFreund awesome, however, it requires creation of a manual helper, no?

**Comment by OnFreund** [comment]
It requires a separate helper, but there's a UI to add and maintain it, and it appears within your device. I use it for water consumption and it's working really well.

**Comment by GuyKh** [comment]
One thing to mention, in this component, there's no way to use it witg the LongTermStatistics - i.e. Energy dashboars as far as I understand

**Comment by OnFreund** [comment]
UtilityMeter supports long term statistics

**Comment by maorcc** [comment]
@OnFreund / @GuyKh, I think it would be best to integrate with a source like [Kamaze](https://www.kamaze.co.il/compare/52/electrical-power#ComparisonPanel) and get all possible plans data. Let the user select the current plan, and based on the last month's consumption pattern, the system can even suggest a better plan if there is such.

**Comment by GuyKh** [comment]
@maorcc while it's not a bad idea, I think the implementation is rough since [kamaze.co.il](https://kamaze.co.il) doesn't have a valid API and the values are textually vague and require parsing - i.e. `הנחה בימים א'-ש' בשעות 23:00 ועד 7:00 למחרת` - what's the discount rate? how to set it into a valid json structure instead of parsing Hebrew each time;

Another maybe more interesting option would be [Chashmalink](https://chashmalink.com/) (by @sinaini) Need to dig deeper into their API

**Comment by OnFreund** [comment]
Also worth noting that HA doesn't accept scraping-based integrations, so using kamaze would make it impossible to contribute this integration into core.
In any case, if you are going for it, I would recommend decoupling the two - one integration for Kamaze that exposes the relevant sensors, and one for IEC that lets you select sensors for the different tariffs.

**Reply by GuyKh** [reply]
I agree about the separation of purposes

**Comment by roysha1** [comment]
Is there any way to apply those with the new release?

**Reply by GuyKh** [reply]
@OnFreund described [UtilityMeter](https://www.home-assistant.io/integrations/utility_meter/#advanced-configuration) for this

**Reply by GuyKh** [reply]
Actually - @OnFreund or anyone else - I'd love to hear the implementation details of Utility Meter combined with IEC sensors - i.e. how did you configured it? which sensor did you use?

**Reply by OnFreund** [reply]
I'm not using the IEC integration yet, but the Utility Meter docs have a [section on how to use tariffs](https://www.home-assistant.io/integrations/utility_meter/#advanced-configuration). The gist is:
* You define the tariff names along with the utility meter
* You set up automations to transition between tariffs (in this case it would be time based, but you can use any trigger that makes sense)
* If you want, you can set up a template sensor to sum up the products of consumption and rates (or multiple sensors, one for each tariff, if you want that resolution)
